atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/ClusterTopologyValidator.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra;
+
+import com.github.rholder.retry.RetryException;
+import com.github.rholder.retry.Retryer;
+import com.github.rholder.retry.RetryerBuilder;
+import com.github.rholder.retry.StopStrategies;
+import com.github.rholder.retry.WaitStrategies;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Sets;
+import com.palantir.atlasdb.keyvalue.cassandra.pool.CassandraServer;
+import com.palantir.conjure.java.api.config.service.HumanReadableDuration;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeRuntimeException;
+import com.palantir.logsafe.logger.SafeLogger;
+import com.palantir.logsafe.logger.SafeLoggerFactory;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import one.util.streamex.EntryStream;
+import org.apache.thrift.transport.TTransportException;
+import org.immutables.value.Value;
+
+public final class ClusterTopologyValidator {
```

Comment:
Originally this class was a utility class but I think it's better to keep it as its own separate object. This ideal for a two reasons:
(1) Makes future refactors much easier. I.e., we can probably make all of these host validation checks a generic interface, and this would be less lift to do if everything was object oriented
(2) Testing becomes much much easier, as you do not need to mock the underlying cassandra clients. Also testing static methods across classes is asking for world of pain.